### PR TITLE
WP beta Stories

### PR DESCRIPTION
**Pull request by GunkaArtur**
Related issue: https://github.com/bagrinsergiu/blox-editor/issues/25400

### CODE REVIEW

File: public/editor-client/src/api/index.ts [hunk 1]
```diff
@@ -1384,4 +1389,76 @@ export const getDefaultLayoutData = async (
...
+  const response = await request(fullUrl, {
+    method: "GET"
+  });
+
+  if (response.ok) {
+    const res = await response.json();
+
+    const parsedResult = mPipe(pass(isStoryDataResponse), (r) =>
+      JSON.parse(r.collection)
+    )(res);
```
Comment (victor-pg) [inline public/editor-client/src/api/index.ts:1434]: apply the same change as in review for cloud api-client

File: public/editor-client/src/defaultTemplates/index.ts [hunk 2]
```diff
@@ -149,46 +152,61 @@ const defaultPopups = (
...
-              thumbnailSrc: `${storiesUrl}/thumbs/${story.pages[0].id}.jpg`,
-              pages: story.pages.map((page) => {
-                return {
-                  ...page,
-                  thumbnailSrc: `${storiesUrl}/thumbs/${page.id}.jpg`
-                };
-              })
-            };
-          })
+          stories: convertStories(templates, `${templatesImageUrl}`),
```
Comment (victor-pg) [inline public/editor-client/src/defaultTemplates/index.ts:172]: here is the same story with `${templatesImageUrl}`

should be used simple: stories: `convertStories(templates, templatesImageUrl)`

File: public/editor-client/src/defaultTemplates/utils.ts [hunk 3]
```diff
@@ -185,6 +231,18 @@ export const isDefaultBlock = (obj: unknown): obj is DefaultBlock => {
   );
 };
 
+export const isDefaultBlockArray = (
+  obj: unknown
+): obj is DefaultBlockWithID[] => {
+  return Array.isArray(obj) && obj.every((item) => isDefaultBlock(item));
```
Comment (victor-pg) [inline public/editor-client/src/defaultTemplates/utils.ts:237]:
```ts
return Array.isArray(obj) && obj.every(isDefaultBlock);
```

File: public/editor-client/src/types/DefaultTemplate.ts [hunk 4]
```diff
@@ -280,3 +271,60 @@ export interface LayoutsPagesResult {
...
+  thumbnailWidth: number;
+  thumbnailHeight: number;
+}
+
+export interface StoryPages {
+  slug: string;
+  thumbnailHeight: number;
+  thumbnailWidth: number;
+  thumbnail: string;
+}
```
Comment (victor-pg) [inline public/editor-client/src/types/DefaultTemplate.ts:291]: create common type with repated keys and use it as extend for `StoriesAPI` and `StoryPages`

**Review by victor-pg — CHANGES_REQUESTED** [review]

**Review by maxval1 — CHANGES_REQUESTED** [review]

**Pull request merged**